File: app/src/main/java/com/example/epigram/data/api/EpigramService.java
package com.example.epigram.data.api;

import androidx.annotation.Nullable;
import com.example.epigram.data.templates.PostTemplate;
import com.example.epigram.data.templates.SearchResult;
import com.example.epigram.data.templates.Wrapper;
import io.reactivex.Single;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;

public interface EpigramService {

    @GET("ghost/api/v2/content/posts")
    Single<Wrapper<PostTemplate>> getPostsFilter(@Query("key") String key, @Query("include") String include, @Nullable @Query("filter") String fiter, @Query("limit") String limit, @Query("page") int page, @Query("order") String order);

    @GET("ghost/api/v2/content/posts")
    Single<Wrapper<PostTemplate>> getPostsBreak(@Query("key") String key, @Query("include") String include, @Nullable @Query("filter") String fiter, @Query("limit") String limit, @Query("order") String order);

    @GET("ghost/api/v2/content/posts")
    Single<Wrapper<SearchResult>> getSearchIDs(@Query("key") String key, @Query("include") String include, @Query("limit") String theLimit, @Nullable @Query("fields") String fields, @Query("page") int page, @Query("order") String order);

    @GET("ghost/api/v2/content/posts/{id}")
    Single<Wrapper<PostTemplate>> getPostFromNotification(@Path("id") String id, @Query("key") String key, @Query("include") String include);

}

File: app/src/main/java/com/example/epigram/data/api/EpigramService.kt
package com.example.epigram.data.api

import com.example.epigram.data.templates.PostTemplate
import io.reactivex.Single
import retrofit2.http.GET
import retrofit2.http.Path

interface EpigramService {

    @GET("/all/page/{page}")
    fun all(@Path("page") page: Int): Single<List<PostTemplate>>

    @GET("/search/{searchTerm}")
    fun search(@Path("searchTerm") searchTerm: String): Single<List<PostTemplate>>

    @GET("/tag/{tag}")
    fun tag(@Path("tag") tag: String): Single<List<PostTemplate>>

    @GET("/article/{id}")
    fun article(@Path("id") id: String): Single<PostTemplate>
}
